bemani/backend/gitadora/nextage.py

```diff
@@ -0,0 +1,1568 @@
+                skilldata.add_child(Node.s32('old_skill',skilldata_dict.get_int('dm_skill')))
+                skilldata.add_child(Node.s32('old_all_skill',skilldata_dict.get_int('gf_all_skill')+skilldata_dict.get_int('dm_all_skill')))
+
+        return root
+
+    def handle_nextage_gametop_request(self, request: Node) -> Node:
+        refid = request.child_value('player/refid')
+        root = self.get_profile_by_refid(refid)
+        if root is None:
+            root = Node.void('nextage_gametop')
+        return root    
+
+    def handle_nextage_gameend_request(self, request: Node) -> Node:
+        refid = request.child_value('player/refid')
+        # Profile save but you can also save scores here
+        if refid is not None:
+            userid = self.data.local.user.from_refid(self.game, self.version, refid)
+        else:
+            userid = None
+        if userid is not None:
+            oldprofile = self.get_profile(userid)
+            newprofile = self.unformat_profile(userid, request, oldprofile)
+        else:
+            newprofile = None
+
+        if userid is not None and newprofile is not None:
+            self.put_profile(userid, newprofile)
+        
+        root = self.get_profile_by_refid(refid)
+        if root is not None:
+            root = Node.void('nextage_gameend')
+            gamemode = Node.void('gamemode')
+            root.add_child(gamemode)
+            gamemode.set_attribute('mode','game_mode')
+            player = Node.void('player')
+            root.add_child(player)
+            player.set_attribute('no','1')
+            skill = Node.void('skill')
+            player.add_child(skill)
+            skill.add_child(Node.s32('rank',1))
+            skill.add_child(Node.s32('total_nr',1))
+            all_skill = Node.void('all_skill')
+            player.add_child(all_skill)
+            all_skill.add_child(Node.s32('rank',1))
+            all_skill.add_child(Node.s32('total_nr',1))
+        if root is None:
+            root = Node.void('nextage_gameend')
+        return root        
+
+    def handle_bemani_gakuen_request(self, request: Node) -> Node:
+        return Node.void('bemani_gakuen')
+
+    def handle_nextage_playablemusic_request(self, request: Node) -> Node:
+        # TO DO: i need to finish all the music import db first.
+        root = Node.void('nextage_playablemusic')
+        # hot
+        hot = Node.void('hot')
+        root.add_child(hot)
+        hot.add_child(Node.s32('major',1))
+        hot.add_child(Node.s32('minor',0))
+
+        # musicinfo. should been import the musicdb at first.
+        # gitadora's music difficulties have changed a lot between each version. 
+        # thus have to collect full music difficulties from each data's musiclist.
+        if self.omnimix == True:
+            all_songs = MUSICLIST_NEXTAGE_OMNIMIX
+        else:
+            all_songs = MUSICLIST_NEXTAGE
```

Comment:
Not sure why these aren't loaded from data.local.music.

Reply:
because musicdb doesnt include all hot music ids of each version. i have to grab  the hot music from bwiki for each one. also its the game gf_skills and dm_skills issues.

Reply:
How much of the info is not present in the game's music DB? Trying to get a handle on what the issue is. Nothing else in the network requires a music list like this so its suspicious (and I don't want to import the code as-is). Some other games like Jubeat are missing some info so they include a TSV file for augmenting the game data. Can this be done similar?

Reply:
hot music items for each version of gitadora.  jubeat tsv is used to replace correct names and artists that are not shown in musicdb.xml. But hot music items is different for each version of gitadora, also hot music affects the player's skill score

Reply:
The musiclist looks like a complete list though, not just a list of song IDs included in the hot list.